src/deno_json/mod.rs

```diff
@@ -914,16 +915,76 @@ impl ConfigFile {
   /// Parse `compilerOptions` and return a serde `Value`.
   /// The result also contains any options that were ignored.
   pub fn to_compiler_options(&self) -> Result<ParsedTsConfigOptions, AnyError> {
-    if let Some(compiler_options) = self.json.compiler_options.clone() {
-      let options: serde_json::Map<String, Value> =
-        serde_json::from_value(compiler_options)
-          .context("compilerOptions should be an object")?;
+    let compiler_options_exists = self.json.compiler_options.is_some();
+    let runtime_exists = self.json.runtime.is_some();
+
+    if compiler_options_exists || runtime_exists {
+      let mut options: serde_json::Map<String, Value> =
+        if let Some(compiler_options) = self.json.compiler_options.clone() {
+          serde_json::from_value(compiler_options)
+            .context("compilerOptions should be an object")?
+        } else {
+          serde_json::Map::new()
+        };
+
+      if runtime_exists {
+        self.resolve_runtime_libs(&mut options);
+      }
+
       parse_compiler_options(options, Some(&self.specifier))
     } else {
       Ok(Default::default())
     }
   }
 
+  /// Resolves and updates the `lib` field in `compilerOptions` based on the `runtime` configuration.
+  fn resolve_runtime_libs(&self, options: &mut serde_json::Map<String, Value>) {
+    if let Some(runtime) = self.json.runtime.clone() {
+      let contains_deno = runtime.contains(&"deno".to_string());
+      let contains_browser = runtime.contains(&"browser".to_string());
+
+      if contains_deno && contains_browser {
+        self.insert_libs(
+          options,
+          vec![
+            "deno.ns".to_string(),
+            "esnext".to_string(),
+            "dom".to_string(),
+            "dom.iterable".to_string(),
+          ],
+        );
+      } else if contains_deno {
+        self.insert_libs(options, vec!["deno.ns".to_string()]);
```

Comment:
This needs to be verified with TSC settings in `deno`, I think we enable more libs for Deno type checking.

Reply:
Sure,
Let me know what other libs we need to set or point me so I find myself.